packages/legend-studio/src/components/editor/edit-panel/mapping-editor/OperationSetImplementationEditor.tsx

```diff
@@ -65,7 +65,11 @@ export const OperationSetImplementationEditor = observer(
       mapping,
       setImplementation.class.value,
     )
-      .filter((si) => si.id.value !== setImplementation.id.value)
+      .filter(
+        (si) =>
+          // filter out set impls already included through a container or the actual container itself
```

Comment:
Think you forgot the case that we can select the same set implementation

Reply:
I didn't since it was included in `leafSetImplementations_withContainers`, but I will change the logic because `leafSetImplementations_withContainers` shouldn't return it.